core/src/main/scala/kafka/consumer/BaseConsumer.scala

```diff
@@ -51,15 +63,63 @@ class NewShinyConsumer(topic: Option[String], whitelist: Option[String], consume
   import scala.collection.JavaConversions._
 
   val consumer = new KafkaConsumer[Array[Byte], Array[Byte]](consumerProps)
-  if (topic.isDefined)
-    consumer.subscribe(List(topic.get))
-  else if (whitelist.isDefined)
-    consumer.subscribe(Pattern.compile(whitelist.get), new NoOpConsumerRebalanceListener())
-  else
-    throw new IllegalArgumentException("Exactly one of topic or whitelist has to be provided.")
-
+  consumerInit()
   var recordIter = consumer.poll(0).iterator
 
+  def consumerInit() {
+    if (topic.isDefined)
+      consumer.subscribe(List(topic.get))
+    else if (whitelist.isDefined)
+      consumer.subscribe(Pattern.compile(whitelist.get), new NoOpConsumerRebalanceListener())
+    else
+      throw new IllegalArgumentException("Exactly one of topic or whitelist has to be provided.")
+  }
+
+  override def receive(): BaseConsumerRecord = {
+    if (!recordIter.hasNext) {
+      recordIter = consumer.poll(timeoutMs).iterator
+      if (!recordIter.hasNext)
+        throw new ConsumerTimeoutException
+    }
+
+    val record = recordIter.next
+    BaseConsumerRecord(record.topic,
+                       record.partition,
+                       record.offset,
+                       record.timestamp,
+                       record.timestampType,
+                       record.key,
+                       record.value)
+  }
+
+  override def stop() {
+    this.consumer.wakeup()
+  }
+
+  override def cleanup() {
+    this.consumer.close()
+  }
+
+  override def commit() {
+    this.consumer.commitSync()
+  }
+}
+
+class NewShinySimpleConsumer(topic: String, partitionId: Int, offset: Long, consumerProps: Properties, timeoutMs: Long = Long.MaxValue) extends NewShinyConsumer(Some(topic), None, consumerProps, timeoutMs) {
+  import org.apache.kafka.clients.consumer.KafkaConsumer
+
+  import scala.collection.JavaConversions._
+
+  override def consumerInit() {
```

Comment:
Hmm, seems `consumerInit` is the only thing that changes and in a pretty straightforward way. Would it make sense to just pass in `partitionId: Option[Int], offset: Option[Long]` and remove the `NewShinySimpleConsumer` class?